Source/MacProj.cpp

```diff
@@ -720,8 +720,11 @@ MacProj::mac_sync_compute (int                   level,
                 const auto& Q = (do_mom_diff == 1 and comp < AMREX_SPACEDIM) ? momenta : Smf;
 
                 amrex::Gpu::DeviceVector<int> iconserv;
+                Vector<int> iconserv_h;
                 iconserv.resize(1, 0);
```

Comment:
You only need to do `conserv.resize(1)`.  That will make the device vector data uninitialized.  But you don't need to.

Reply:
You think it could have a noticeable effect even with all the projections?

Reply:
I'll update that !